conda/core/prefix_data.py

```diff
@@ -320,8 +316,9 @@ def load(self):
             )
             for meta_file in conda_meta_json_paths:
                 self._load_single_record(meta_file)
-        if self._pip_interop_enabled:
-            self._load_site_packages()
+        if self._pip_interop_enabled:  # TODO: Rename as general interop
```

Comment:
Alternatively (or in addition) the interoperability could be toggled per plugin?

Reply:
Yes but not at this API level, I think. Interoperability can be used in two very different ways:

- To report extra stuff in `conda list`
- To enable the solver to "see" more packages as "installed" and consider them as part of the solve process

For (1) we can have as much interoperability as we want because it's harmless, but for (2) it can cause a lot of havoc. So I was more leaning towards having a global bool switch `with_loader_plugins`. Then each plugin can be enabled / disabled via config I guess? It would also bring some trouble with the metaclass cache stuff...

But for now, we only need this part. I can rename it now, though.